modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/importer/type/CSVCommerceOrderImporterTypeImpl.java

```diff
@@ -204,13 +227,42 @@ protected void activate(Map<String, Object> properties) {
 	}
 
 	private CommerceOrderImporterItemImpl[] _getCommerceOrderImporterItemImpls(
-			long companyId, long commerceChannelGroupId, FileEntry fileEntry)
+			long companyId, long commerceChannelGroupId,
+			FDSPagination fdsPagination, FileEntry fileEntry)
 		throws Exception {
 
+		List<CSVRecord> csvRecords = new ArrayList<>();
+
 		CSVParser csvParser = _getCSVParser(fileEntry);
 
+		if (fdsPagination == null) {
```

Comment:
So we are "reading" the CSV file from "start" to "end" based on the pagination. This way we need to call this one every time we change page. Isn't it beter to load the entire file in memory (CommerceOrderImporterItemImpl[]) and paginate just the array to return a subset of it?

Reply:
yeah it would be better performance to just do a initial load once, let me try this out.